Create drivetrain

FRCTeam3255-Shared-K1-10 opened the pull request.

FRCTeam3255-Shared-K1-10 committed:
Declare Drivetrain Motors

Created a subsystem and added drivetrain motors and motor id's.

FRCTeam3255-Shared-K1-10 committed:
Added myself as a contributor

Co-Authored-By: Evan Grinnell <155865196+S0L0GUY@users.noreply.github.com>

FRCTeam3255-Shared-K1-10 committed:
Merge branch 'main' into create-drivetrain

FRCTeam3255-Shared-K1-10 committed:
Added Motor Movement Logic

Co-Authored-By: Evan Grinnell <155865196+S0L0GUY@users.noreply.github.com>

FRCTeam3255-Shared-K1-10 committed:
Added Empty Drive Command

S0L0GUY committed:
Created Basic Driving Control Sample

This code does not work correctly, but that is to be fixed in the next push.

FRCTeam3255-Shared-K1-10 committed:
Made drivetrain turn and not angry

Co-Authored-By: Evan Grinnell <155865196+S0L0GUY@users.noreply.github.com>

FRCTeam3255-Shared-K1-10 committed:
Merge branch 'main' into create-drivetrain

TaylerUva commented:
Good except one bug:
https://github.com/FRCTeam3255/2024_Sportbot_Code/actions/runs/11533884009/job/32107301309?pr=7#step:6:39

S0L0GUY committed:
Fixed Errors and Simplified Code

Added drivetrain to the requirements, made all of the talon motors private, and compressed two functions into one for more readable code.

TaylerUva commented on src/main/java/frc/robot/commands/Drive.java:
```diff
@@ -0,0 +1,47 @@
+  // Called every time the scheduler runs while the command is scheduled.
+  @Override
+  public void execute() {
+    globalDrivetrain.setDrivetrainSpeed(globalForwardSpeed.getAsDouble() - globalRotationSpeed.getAsDouble(), globalForwardSpeed.getAsDouble() + globalRotationSpeed.getAsDouble());
```
I would move the math into the method

S0L0GUY committed:
Moved Rotation Math to 'Drivetrain.java'

TaylerUva commented on src/main/java/frc/robot/subsystems/Drivetrain.java:
```diff
@@ -23,8 +23,26 @@ public Drivetrain() {
+   * @param leftVelocity The velocity to set for the left side of the drivetrain.
+   * @param rotationSpeed The rotation speed to apply to the drivetrain.
+   */
+  public void setDrivetrainSpeed(double rightVelocity, double leftVelocity, double rotationSpeed) {
```
Why not just forward velocity and rotate velocity?

S0L0GUY replied:
That makes sence

S0L0GUY committed:
Compressed Arguments in 'setDrivetrainSpeed'

TaylerUva reviewed (approved):
Excellent work!

TaylerUva commented:
@S0L0GUY looks like there are merge conflicts

S0L0GUY committed:
Merge branch 'main' into create-drivetrain

S0L0GUY committed:
Resolved Errors Created During Merge

S0L0GUY committed:
Re-merged `RobotContainer.java`

"Hopefully" correctly merged `RobotContainer.java` with the old version. ㅠㅜ
